dd-sdk-android/src/main/kotlin/com/datadog/android/rum/internal/domain/RumEventSerializer.kt

```diff
@@ -0,0 +1,131 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-2020 Datadog, Inc.
+ */
+
+package com.datadog.android.rum.internal.domain
+
+import com.datadog.android.core.internal.domain.Serializer
+import com.google.gson.JsonArray
+import com.google.gson.JsonNull
+import com.google.gson.JsonObject
+import com.google.gson.JsonPrimitive
+import java.text.SimpleDateFormat
+import java.util.Date
+import java.util.Locale
+import java.util.TimeZone
+
+internal class RumEventSerializer : Serializer<RumEvent> {
+
+    private val simpleDateFormat = SimpleDateFormat(ISO_8601, Locale.US).apply {
+        timeZone = TimeZone.getTimeZone("UTC")
+    }
+
+    // region Serializer
+
+    override fun serialize(model: RumEvent): String {
+        val root = JsonObject()
+        val view = JsonObject()
+        val event = JsonObject()
+        val rum = JsonObject()
+        val http = JsonObject()
+        val resource = JsonObject()
+
+        // Event Context
+        val context = model.context
+        root.addProperty(TAG_APPLICATION_ID, context.applicationId.toString())
+        root.addProperty(TAG_SESSION_ID, context.sessionId.toString())
+        view.addProperty(TAG_VIEW_ID, context.viewId.toString())
```

Comment:
so all the events are linked through these 3 ids which are unique per session ? Means that for each Page (Activity, Fragment,etc) we will have to keep a map with the page url -> uniqueID to know what's the unique id generated for that page whenever the page is detected right ?

Reply:
I think we will need a Session class where to keep all these IDs maps and which should be re - generated every time there's a new session started

Reply:
The application will be a constant (like the token)
Session ID will be handled and unique per session
View id will be unique per view (= 1 per activity run), you'll see that there's a map in the DatadogRumMonitor to track the known active views.